tests/test_magpie_ui.py

```diff
@@ -298,6 +298,312 @@ def check_api_resource_permissions(resource_permissions):
+        resp = utils.test_request(self, "GET", "/ui/register/users")
+        data = {
+            "user_name": test_register_user,
+            "password": test_register_user,
+            "confirm": test_register_user,
+            "email": test_register_email,
+        }
+
+        # custom app settings for this test
+        settings = {
+            "magpie.user_registration_enabled": True,
+            "magpie.user_registration_approval_enabled": True,
+            "magpie.user_registration_notify_enabled": True,
+            "magpie.user_registration_approval_email_recipient": "fake-admin@test.com",
+            "magpie.user_registration_notify_email_recipient": "notify@user-registration.com",
+        }
+
+        with utils.mocked_get_settings(settings=settings):
+            with utils.mock_send_email() as email_contexts:
+                _, wrapped_contents, mocked_send = email_contexts
+
+                # submit the registration form to trigger the confirmation email
+                resp = utils.TestSetup.check_FormSubmit(self, previous_response=resp, form_data=data,
+                                                        form_match="add_user_form", form_submit="create")
+                utils.check_ui_response_basic_info(resp, expected_title="Magpie User Registration")
+                utils.check_val_equal(mocked_send.call_count, 1,
+                                      msg="Expected sent notification to user for its email validation.")
+
+                # simulate user clicking the confirmation link in 'sent' email (external operation from Magpie)
+                confirm_url = wrapped_contents.call_args.args[-1].get("confirm_url")
+                resp = utils.test_request(self, "GET", urlparse(confirm_url).path)
+                body = utils.check_ui_response_basic_info(resp, 200, expected_title="Magpie User Registration")
+                utils.check_val_is_in("email has been confirmed", body)
+
+                # retrieve the decline endpoint from the 'sent' email and authenticate as admin
+                # simulate the administrator clicking the decline link to suppress the pending approval
+                utils.check_val_equal(mocked_send.call_count, 2,
+                                      msg="Expected email to have been sent to the administrator for approval.")
+                decline_url = wrapped_contents.call_args.args[-1].get("decline_url")
+                decline_path = urlparse(decline_url).path
+                self.login_admin()  # admin needed for decline link
+                resp = utils.test_request(self, "GET", decline_path)
+                body = utils.check_ui_response_basic_info(resp, 200)
+                utils.check_val_is_in("Pending user registration was successfully declined", body)
+                utils.check_val_equal(
+                    mocked_send.call_count, 2,
+                    msg="neither pending user nor notify emails should be sent since registration was declined"
```

Comment:
Is there a way for the pending user to know that his registration was declined?

Reply:
No. It was decided during previous discussions that declines should just be discarded. 
Unless @dbyrns wants to revert that decision such that we also send an email about the decline similarly to the approved email.

Reply:
I will let you guys take the final decision. But for a user, that has confirmed his registration and is waiting for the admin approval, it could be good to notify him (via email as you suggest) of a request denial instead of leaving him in the unknown.

Reply:
We discussed essentially about the use case where admins would generally approve everyone unless something is suspicious. In that case, decline mail seems indeed superfluous. In a case where we could have a good reason to decline some people (by using a particular domain email for example), it could be helpful for admin to create a decline mail template explaining the guideline under which registration could be refused.

Reply:
There is no way currently to input some kind of "reason" that the admin could provide to decline the email, just to keep the decline operation simple enough to be a one-button-click operation.
So unless, the template is extremely specific, it will rarely/never cover all cases of potential decline.

Reply:
I agree. I don't see that as a "this is why I refuse you", but more a "we accept that kind of requests". Maybe the template will not fit the exact case, but at least the user will know that its request as been declined. It also allows to put an email to reach an administrator if wanted.

Reply:
see latest commits for added decline template email